pitch.md

```diff
@@ -1,44 +1,90 @@
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain? Be careful
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+I think the language is almost definitely going to have to be an external DSL - it doesn't really make sense to me to have it as an internal DSL, since its syntax is so different from Scala's (the language I plan to build my project in). In addition, after completing the assignments piconot and piconot-external, I have a feeling that it might be easier for me to implement this language as an external DSL - it felt like Scala's features hindered more than helped when we were implementing Picobot as an internal DSL, and I see no reason why something different would happen for this language. 
+
+For CS111, I'm going to make the language interactable only through a text-based program. Programming would involve typing commands in the syntax of the language. I think this is the right way to create the language as a project for CS111 because creating a GUI (though it may be more user friendly) would be adding an additional level of complication that doesn't have much to do with language design, which is meant to be the focus of this project. If you wanted to make this language more usable for the general public, though, it would probably be advantageous to build a GUI for it eventually.
+
+A program might look something like this:
+```
+Begin LATTE:
+1. 2 shots espresso
+2. Add 10 oz 2% milk
+3. Add 2 oz foam
+4. Add 4 pumps sugar
+End LATTE:
+```
+
+Or like this:
+
+```
+Begin PUMPKIN_SPICE_LATTE:
+1. LATTE
+2. Add 2 pumps pumpkin
+3. Add 1 shake Cinnamon
+~~4. Add judgement~~
+End PUMPKIN_SPICE_LATTE
+```
+
+Note: Here, you can also see a "function call" being executed (PUMPKIN_SPICE_LATTE calls LATTE). This is a feature that I'd like to build into the language. 
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+In an ideal world, the result of running a program in this language would be a delicious drink. However, since I don't have the resources to procure an actual automated coffee machine to program using this lanugauge, I will have to come up with some alternative "final product" of the language. I'm currently leaning towards building an API that represents the coffee machine. API calls would result in printed statements like "pulling *n* shots of espresso now". Thus, the final product of a program will just be a bunch of print statements, but the idea is that if you swapped out the print statments for actual implementation of the automated coffee making, you could keep the API the same and the code that the user writes to create the drink wouldn't have to be changed at all.
+
```

Comment:
It would probably have a different format. After talking to Prof Ben, I revised my idea for what to do for the backend somewhat - it would be cool to have multiple backends, to show that the language is truly portable. So, for instance, in one mode it might print the steps it executes to make the drink, in another it may analyze the recipe for the quantity of ingredients required, in another it may produce a visual representation of the drink, etc.